sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -802,10 +812,14 @@ synchronized void setLatestOffset(long latestOffset) {
 
       synchronized long approxBacklogInBytes() {
         // Note that is an an estimate of uncompressed backlog.
+        return (long) (backlogMessageCount() * avgRecordSize);
```

Comment:
Should return `UnboundedReader.BACKLOG_UNKNOWN` when backlogMessageCount returns the same instead of multiplying it?